add config file to input step selections and screen parameters

hansen508 opened the pull request:
Changes:
- add config.ini with comments/defaults for each choice
- add function to parse config file in setup_misosoupy
- call config function in run_misosoupy, psychopy_present_item_list, and psychopy_refine_item_list
- remove redundant input arguments from functions, shorten docstrings

I have a list of optimizations I'm still working on (in addition to documenting sample usage), but I want to make sure this doesn't blow up on your end before I keep going.

hansen508 committed:
add config file to input step selections and screen paramters

- add config.ini with comments/defaults for each choice
- add function to parse config file in setup_misosoupy
- call config function in run_misosoupy, psychopy_present_item_list, and psychopy_refine_item_list
- remove redundant input arguments from functions, shorten docstrings

hansen508 committed:
fix step_refine code within organization step

hansen508 committed:
add errors if selected setup choices are incompatible

hansen508 committed:
split functions for presenting instructions and exiting out into separate scripts

hansen508 committed:
add option to change number of sounds participants need to select

hansen508 committed:
update instructions depending on which steps are chosen

hansen508 committed:
add check for if participant ID already exists, prompt to choose a different ID if so

hansen508 committed:
fix number of required sounds to select if there are fewer options available

hansen508 committed:
change back config from testing

danibene commented on src/misosoupy/psychopy_present_item_list.py:
```diff
@@ -8,34 +8,35 @@
+# Import config file and screen parameters
+import psychopy_exit_out
+import setup_misosoupy
+config_path = setup_misosoupy.get_home_dir() + os.sep + 'config.ini'
```
You could have a function dedicated to loading the config file to avoid repetition across python files, and in case anyone wants to change the path at some point it would be easier

hansen508 replied:
ooh okay yes, do you think I should just store the config file in the assets directory, so the path gets loaded in the get_path_to_assets function? Or make a get_path_to_config function with similar code?

Or, alternatively, maybe I just define the path to the config file directly in setup_misosoupy, and remove it as a necessary input when calling the function? So that way I wouldn't need to redefine the path across python files?

danibene replied:
I think these are all options! Even if you move it to the assets directory I would still either create a function like get_path_to_config or defining it as a constant in setup_misosoupy (you can then import that variable).

danibene commented on src/misosoupy/psychopy_refine_item_list.py:
```diff
@@ -45,22 +47,14 @@ def function_present_refined_item_list(
     mean_length : int
         Average number of characters comprising the sound labels. Used to determine font size.
     setup_item_height : int
         Font height. Default is 0.085.
```
I think I would either specify that this default is defined in the config.ini file or put a default value here in the function too

danibene commented on src/misosoupy/setup_misosoupy.py:
```diff
@@ -76,3 +78,32 @@ def get_path_to_assets():
+        else:    
+            try:
+                screen_parameters[key] = float(value) 
+            except:
```
Usually bare excepts are discouraged, I would try to at least specify the error (and maybe there is a way to avoid a try and except completely): https://python101.pythonlibrary.org/chapter7_exception_handling.html#:~:text=In%20Python%2C%20this%20is%20known,a%20division%20by%20zero%20error.

hansen508 replied:
helpful, thanks!

danibene reviewed (approved):
This is great, feel free to merge!

hansen508 committed:
specify default for num_items_to_select

Per comment https://github.com/miso-sound/misosoupy/pull/15#discussion_r1739945917

hansen508 committed:
specify exception error

Per comment https://github.com/miso-sound/misosoupy/pull/15#discussion_r1739946022
